geoprocessing/src/utils/file.utils.ts

```diff
@@ -25,3 +29,36 @@ export const uploadOptions: Readonly<MulterOptions> = {
     },
   }),
 };
+
+export class FileService {
+  unzipFile(fileInfo: Express.Multer.File): Promise<string> {
+    return new Promise((resolve, reject) => {
+      createReadStream(fileInfo.path)
+        .pipe(
+          Extract({
+            path: path.join(
+              fileInfo.destination,
+              path.basename(fileInfo.filename.replace('.zip', '')),
+            ),
+          }),
+        )
+        .on('close', () =>
+          resolve(`${fileInfo.filename} extracted successfully`),
+        )
+        .on('error', (error: Error) =>
+          reject(
+            new Error(`${fileInfo.filename} could not be extracted: ` + error),
+          ),
+        );
+    });
+  }
+
+  deleteDataFromFS(path: string): void {
+    if (path.startsWith('/tmp')) {
+      unlink(path);
+      rmdir(path.replace('.zip', ''), { recursive: true });
```

Comment:
it should be awaited, and usages of `deleteDataFromFS` should at least handle rejections

Reply:
@Dyostiq 1105fe35e9b4ce2eeb8265172f46496f055d8978